---
# Documentation: https://wowchemy.com/docs/managing-content/

title: "A comparative advantage logic"
subtitle: ""
summary: ""
authors: [admin] 
tags: [perdagangan, ekonomi]
categories: []
date: 2025-01-06T11:00:00+07:00
lastmod: 2025-01-06T11:00:00+07:00
featured: false
draft: false

# Featured image
# To use, add an image named `featured.jpg/png` to your page's folder.
# Focal points: Smart, Center, TopLeft, Top, TopRight, Left, Right, BottomLeft, Bottom, BottomRight.
image:
  caption: ""
  focal_point: ""
  preview_only: false

# Projects (optional).
#   Associate this post with one or more of your projects.
#   Simply enter your project's folder or file name without extension.
#   E.g. `projects = ["internal-project"]` references `content/project/deep-learning/index.md`.
#   Otherwise, set `projects = []`.
projects: []
---

Semakin maraknya pembicaraan soal swasembada dan anti-impor semakin kuat aja. Terakhir yang saya sering baca adalah soal [impor paksa sapi](https://www.youtube.com/watch?v=-G2EjOriAjY&t=1s) dan lebih serem lagi, soal [deklarasi tidak akan impor pangan](https://www.tempo.co/ekonomi/zulkifli-hasan-tegaskan-komitmen-soal-swasembada-pangan-stop-impor-beras-hingga-gula-mulai-tahun-ini-1195061). Ntah bagaimana cara pemerintah meningkatkan produktivitas pertanian, yang mana merupakan satu-satunya cara agar bisa memenuhi kebutuhan pangan domestik tanpa impor. Tapi sepertinya sih ketika harga terlihat mulai naik, kemungkinan besar impor bakal tetap dibuka.

Terlepas dari dinamika ini, saya rasa cukup jelas bahwa pernyataan model gitu tidak didasari oleh perhitungan ekonomi yang cukup.

## Ekonomi tertutup

Mari kita buat sebuah model kecil untuk ekonomi sebuah negara dengan aturan main sebagai berikut:

1. Penduduk negara ini hanya mengonsumsi 2 jenis barang, yaitu pangan (A) dan sandang (B). Kombinasi konsumsi penduduknya mengikuti fungsi $U={A^D}^{0.5}{B^D}^{0.5}$. Artinya, kombinasi konsumsi A dan B mana yang akan memaksimalkan kebahagiaan penduduknya, itulah yang dipilih.
2. Faktor produksi di negara ini adalah pekerja (L), di mana $L_A$ adalah pekerja di industri pangan dan $L_B$ adalah pekerja di industri sandang. Fungsi produksinya linear aja, seperti di ricardian economy, yaitu $A^S=f\times L_A$ dan $B^S=g\times L_B$, di mana $f dan g$ adalah parameter positif.
3. Jumlah total tenaga kerja di negara ini adalah $L$ dan pekerja di 2 industri tidak boleh lebih banyak dari ini, jadinya $L_A+L_B=L$.
4. ekonomi tertutup berarti demand harus sama dengan supply, atau $A^S=A^D$ dan $B^S=B^D$ karena kelebihan produksi yang tidak dikonsumsi tidak bisa diekspor, dan kekurangan konsumsi yang tidak diproduksi tidak bisa diimpor[^1].

Dengan kata lain, objective function dari ekonomi kecil kita ini adalah:

$$
\begin{align*}
\max_{A^D,B^D} & {A^D}^{0.5}{B^D}^{0.5} \\
\text{subject to} & A^S=f\times L_A \\

\end{align*}
$$



In [15]:
#| warning: false
#| error: false
#| echo: false
import pandas as pd
import numpy as np
from scipy.optimize import linprog
from scipy.optimize import fsolve

class trade:
  r"""
  Model ini bertujuan untuk menunjukkan logika comparative advantage. Model ini sangat simpel, tapi secara konseptual tetap dapat digunakan untuk menganalisis perdagangan barang dan jasa. Ini adalah pintu masuk yang cukup "bersahabat" untuk semua calon analis ekonomi internasional. (memulai dengan model yang mudah juga dilakukan mahasiswa teknik kok!)

  Mari kita asumsikan di dunia ini cuma ada 2 barang, yaitu Apple (A) dan Batubara (B). Asumsikan juga di dunia ini cuma ada 2 negara, yaitu Home (H) dan Foreign (F). Modelnya punya peraturan seperti ini:

  1. kebahagiaan di kedua negara merupakan sebuah fungsi yang memetakan A,B ke sebuah index yang kita namakan U. Fungsinya adalah Ui={A^ai}*{B^bi} di mana i adalah negara, bisa H, bisa F.

  2. Untuk membuat A dan B dibutuhkan tenaga kerja sebanyak LA (jumlah pekerja yang digunakan untuk bikin A) dan LB (jumlah pekerja yang digunakan unuk bikin B). Fungsi produksi A dan B adalah Ai=fi*LAi dan Bi=gi*LBi.

  3. Jumlah tenaga kerja di kedua negara adalah LH dan LF. Dengan kata lain, jumlah tenaga kerja yang bisa digunakan untuk kerja di 2 industri di 2 negara tersebut terbatas. Alias, LH=LAH+LBH dan LF=LAF=LBF

  

  """
  
  def __init__(self, LH=100.0,  
                     LF=100.00,   
                     aH=0.5,         
                     bH=0.5,          
                     aF=0.5,         
                     bF=0.5,              
                     fH=0.25,             
                     gH=0.75,              
                     fF=0.75,
                     gF=0.25
                     
                     ):
    self.LH,self.LF,self.aH,self.bH,self.aF,self.bF,self.fH,self.gH,self.fF,self.gF=LH,LF,aH,bH,aF,bF,fH,gH,fF,gF
    
  def hasil(self):
    LH,LF,aH,bH,aF,bF,fH,gH,fF,gF=self.LH,self.LF,self.aH,self.bH,self.aF,self.bF,self.fH,self.gH,self.fF,self.gF
    # Construct parameters
    c_ex1 = np.array([pa,pb,pg])

    # Inequality constraints
    A_ex1 = np.array([[-1, -1,-1],
                  [a,b,g]])
    b_ex1 = np.array([-omega,e])

    bounds_ex2 = [ba,
                  bb,
                  bg]

    # Solve the problem
    # we put a negative sign on the objective as linprog does minimization
    res_ex1 = linprog(c_ex1, A_ub=A_ex1, b_ub=b_ex1,bounds=bounds_ex2)
    return res_ex1
  
  def biaya(self):
    omega,e,pa,pb,pg,a,b,g=self.omega,self.e,self.pa,self.pb,self.pg,self.a,self.b,self.g
    h=self.hasil()['x']
    v=np.array((pa,pb,pg))
    c=(h @ v)/1e12
    return c
#    return print(f'total biaya pembangkit listrik adalah {c} triliun rupiah')
  
  def emisi(self):
    omega,e,pa,pb,pg,a,b,g=self.omega,self.e,self.pa,self.pb,self.pg,self.a,self.b,self.g
    h=self.hasil()['x']
    v=np.array((a,b,g))
    m=h @ v
#    return print(f'total emisi adalah {m} kgCO2')
    return m
    
  def summary(self):
    omega,e,pa,pb,pg,a,b,g=self.omega,self.e,self.pa,self.pb,self.pg,self.a,self.b,self.g
    m=self.emisi()
    c=self.biaya()
#    lal=print(f'total biaya pembangkit listrik adalah {c:.2f} triliun rupiah')
#    lel=print(f'total emisi adalah {m:,.2f} kgCO2')
    lal=print(f'The total cost is {c:.2f} trillion IDR or {c/omega*1e9:,.2f} IDR/KWh')
    lel=print(f'The total emission is {m:,.2f} kgCO2')
    sumber=('renewables','coal','other fossil fuels')
    itung=self.hasil()['x']
    for i,j in zip(sumber,itung):
        print(f'Total electricity by {i} is {j:,.2f} MWh ({j/omega*100:.2f} %)')
    return lal
    return lel
  
  def cd(self,al=.1,
              be=.6,
              ga=.3,
              b1=1e6,
              b2=1e6,
              b3=1e6,
              ):
    omega,e,pa,pb,pg,a,b,g=self.omega,self.e,self.pa,self.pb,self.pg,self.a,self.b,self.g
    def func(w):
      omuga=108558478
      return [w[0]-omuga*((pb*al/pa/be)**be)*((pg*al/pa/ga)**ga),
              w[1]-omuga*((pa*be/pb/al)**al)*((pg*be/pb/ga)**ga),
              w[2]-omuga*((pa*ga/pg/al)**al)*((pb*ga/pg/be)**be),
              a*w[0]+b*w[1]+g*w[2]-e]
    root = fsolve(func,[b1,b2,b3,1],xtol=0.01)
    root
    return root
  
  def hasilcd(self):
    omega,e,pa,pb,pg,a,b,g=self.omega,self.e,self.pa,self.pb,self.pg,self.a,self.b,self.g
    sumber=('renewables','coal','other fossil fuels')
    itung=(self.cd()[0],self.cd()[1],self.cd()[2])
    c=(pa*self.cd()[0]+pb*self.cd()[1]+pg*self.cd()[2])/1e12
    m=a*self.cd()[0]+b*self.cd()[1]+g*self.cd()[2]
    for i,j in zip(sumber,itung):
        print(f'Total electricity generated from {i} is {j:,.2f} MWh ({j/omega*100:.2f} %)')
    lal=print(f'The total cost of electricity generation is {c:,.2f} trillion IDR')
    lel=print(f'The total emission is {m:,.2f} kgCO2')
    return lal
    return lel
    

tahun     ppn      pdbn     rasio
0   2016  412213  12401728  3.323835
1   2017  480724  13589825  3.537382
2   2018  537267  14838756  3.620701
3   2019  531577  15832657  3.357472
4   2020  450328  15443353  2.915999
5   2021  551900  16976751  3.250917
6   2022  687609  19588089  3.510342
7   2023  742264  20892376  3.552798
8   2024  811365  22084570  3.673900

Jangan lupa bahwa tahun 2020 ada COVID, tahun 2021 recovery, lalu tahun 2022 adalah tahun kenaikan PPN dari 10% ke 11%. `ppn` itu adalah angka PPN+PPNBM dari BPS, sementara `pdbn` adalah PDB nominal. PDB nominal di tahun 2024 adalah perkiraan berdasarkan pertumbuhan PDB nominal tahun sebelumnya, tapi PPN di 2024 sudah fix karena angkanya udah ada di BPS. Angka PDB nominal 2024 sayangnya belum ada di BPS.

Dapat kita lihat bahwa rasio PPN+PPNBM per PDB nominal di 2022 dan 2023 meningkat dibandingkan 2021, sekitar 0.3 percentage points. Tapi jangan lupa bahwa angka ini cukup normal karena di 2017 dan 2018 juga sekitar segitu. 2020 dan 2021 masih bisa kita anggap COVID lah mungkin. yang jelas, peningkatan di 2022 dan 2023 nggak begitu clear.

Tapi let's say kita anggap peningkatan 0.3 percentage points itu beneran berkat naiknya PPN ya. PPN naik dari 10% ke 11%, atau 1 percentage points, berarti naiknya 10% (dari 10% ke 11% berarti naik 1/10). Sementara itu, kenaikan 10% dari 3.25% = 0.32 percentage points, yaaa mirip lah ya sama kenaikan PPN-nya. 

Meski begitu, gw rasa ini gak bisa dibilang konklusif, mengingat ya tadi, 2021 masih recover, dan kita pernah 3.6% PPN+PPNBM per PDB nominal di 2018, padahal waktu itu PPN kan masih 10%. Makanya gue bilang di _press release_ bahwa kenaikannya nggak efektif. Masih di dalam batas 1 standar deviasi, bahkan ketika kita hitung COVID.

Yang lebih penting, recovery Indonesia di 2022 tidak setinggi negara lain, padahal 2022 itu kita lagi mengalami ledakan harga komoditas. Ini mungkin sih bukan gara-gara PPN saja sih, jadi ya masih jauh dari konklusif.

<iframe src="https://data.worldbank.org/share/widget?end=2023&indicators=NY.GDP.MKTP.KD.ZG&locations=ID-MY-PH-VN&start=2016" width='450' height='300' frameBorder='0' scrolling="no" ></iframe>

Tapi sepertinya bagi World Bnak, kenaikan PPN ini sebenernya sudah cukup _justifiable_.

## Bantuan The World Bank
Yes, sebenernya rencana untuk memperbaiki penerimaan dan belanja negara sudah lama digaungkan Bu Sri Mulyani, tepatnya sejak 2019. Hal ini disampaikan di [siaran pers World Bank ini (2022)](https://www.worldbank.org/en/news/press-release/2022/06/24/new-support-will-help-strengthen-indonesia-tax-system-improve-development-spending). Siaran pers ini menyatakan support dari Bank Dunia terhadap program untuk memperbaiki penerimaan dan belanja negara. Support tersebut diwujudkan oleh Bank Dunia dalam bentuk pinjaman lunak \$750 juta untuk membantu pemerintah Indonesia mewujudkan penerimaan yang lebih tinggi, adil dan berkelanjutan, serta memperkuat efisiensi belanja negara. Nama programnya adalah "Indonesia Fiscal Reform Development Policy Loan" dan detailnya bisa baca di [sini](https://projects.worldbank.org/en/projects-operations/project-detail/P177726)

<blockquote class="twitter-tweet"><p lang="in" dir="ltr">Salah satu area fokus dalam kemitraan Bank Dunia dan Indonesia 2021-2025: memperkuat daya saing dan ketahanan ekonomi melalui pengumpulan pajak yang lebih baik dan belanja anggaran pemerintah yang lebih efektif.<br><br>Selengkapnya 👉<a href="https://t.co/Q9zsdxx6Zj">https://t.co/Q9zsdxx6Zj</a> <a href="https://t.co/R6xb06d1Lh">pic.twitter.com/R6xb06d1Lh</a></p>&mdash; World Bank Indonesia (@BankDunia) <a href="https://twitter.com/BankDunia/status/1399614261291368452?ref_src=twsrc%5Etfw">June 1, 2021</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script> 

Bantuan bernilai fantastis itu mendorong 2 pilar, yaitu pilar penerimaan negara dan pilar belanja negara. Hasil reviewnya ([Report nomor ICRR0024178](https://documents1.worldbank.org/curated/en/099063024233521168/pdf/BOSIB1d7b7faba0131a0c91c259cca12295.pdf)) untuk pilar penerimaan negara lebih kurang tergambar di tabel di bawah ini.

![Hasil program World Bank](image.png)

Di 2023, rasio PPN/PDB menurut World Bank adalah 3.7%, lebih tinggi daripada hitungan saya, dan angka ini ada di atas target mereka yaitu 3.4%. Jadi meski angka saya lebih kecil, dia tetap ada di atas target World Bank, yang mana bisa dibilang kenaikan PPN dan penghilangan _exemption_ untuk produk minerba, sudah cukup berhasil menaikkan PPN.

Di samping kenaikan PPN, World Bank juga berhasil membantu Kemenkeu menaikkan PPh orang karena menciptakan _brackett_ baru sembari menaikkan _threshold_ PPh paling bawah, yang berarti lebih progresif. Sayangnya, World Bank gagal membantu Kemenkeu mewujudkan pajak karbon. Berhasil 2 dari 3 sih lumayan banget ya.

Nah yg saya belum clear adalah apakah \$750 juta nya cuma sampai sini atau ada kegiatan lanjutan lagi. Buat yang tertarik menggali lebih dalam, coba aja baca-baca dokumen dari World Bank yang link-nya dah saya taruh. Bagus kok World Bank, lumayan transparan.

Dah mungkin begitu aja detil yang mau saya share di tulisan ini. Semoga bermanfaat.

